model/criteria/criteria_test.go

```diff
@@ -89,4 +90,94 @@ var _ = Describe("Criteria", func() {
 		gomega.Expect(newObj.OrderBy()).To(gomega.Equal("random() asc"))
 	})
 
+	It("extracts all child smart playlist IDs from All expression criteria", func() {
+		var topLevelInPlaylistID = uuid.New().String()
```

Comment:
Nitpick: please use this style:
```suggestion
		topLevelInPlaylistID := uuid.NewString()
```